modules/library/metadata/src/test/java/org/geotools/util/URLsTest.java

```diff
@@ -101,17 +101,17 @@ public void testUrlToFile() throws Exception {
         assertURL(replaceSlashes("C:\\one"), "file://C:/one");
         assertURL(replaceSlashes("C:\\one\\two"), "file://C:/one/two");
         assertURL(replaceSlashes("C:\\one\\two\\and three"), "file://C:/one/two/and three");
-        assertEquals("sample", URLs.urlToFile(new URL("file:sample?query")).toString());
-        assertEquals("sample", URLs.urlToFile(new URL("file:sample#ref")).toString());
+        assertURL("sample", "file:sample?query");
```

Comment:
@vjeetje these `assertEquals` to `assertURL` changes are good. I do not know how we missed these. Thanks!